server/controllers/profile-controller.js

```diff
@@ -19,22 +19,17 @@ class ProfileController {
    */
   static async getProfile(req, res) {
     const { username } = req.params;
+    const { id } = req.user;
     try {
-      return User.findOne({ where: { username } })
-        .then(data => {
-          if (!data) {
-            return errorResponse(res, 'username does not exist', 404);
-          }
-          delete data.dataValues.hash;
-          return successResponse(
-            res,
-            {
-              user: data.dataValues
-            },
-            200
-          );
-        })
-        .catch(() => errorResponse(res));
+      const profile = await User.findOne({
+        where: { username },
+        attributes: ['id', 'username', 'email', 'bio', 'image']
+      });
+      if (!profile) {
+        return errorResponse(res, 'user not found', 404);
+      }
+      const isFollower = await profile.hasFollower(id);
```

Comment:
Please look into whether this can be combined with the first call to the database (`User.findOne`)

Reply:
I will look for a better way to implement it